docs/dev/serialization.md

```diff
@@ -0,0 +1,110 @@
+This developer document explains how Cirq serializes objects into (and out of) JSON.
+It also explains how to add a new serializable object,
+how to remove a serializable object from cirq while maintaing backwards compatibility with old serialized files,
+and various related guidelines.
+
+## Exposed API
+
+Most Cirq objects can be converted into json using the `cirq.to_json` method.
+For example, this code:
+
+```python
+import cirq
+import sympy
+
+obj = cirq.X**sympy.Symbol('t')
+
+print(cirq.to_json(obj))
+```
+
+Prints this output:
+
+```
+{
+  "cirq_type": "XPowGate",
+  "exponent": {
+    "cirq_type": "sympy.Symbol",
+    "name": "t"
+  },
+  "global_shift": 0.0
+}
+```
+
+The JSON can also be written to a file:
+
+```python
+cirq.to_json(obj, filepath)
+```
+
+Or read back in from a file:
+
+```python
+obj = cirq.read_json(file_path)
+```
+
+Or read from a string:
+
+```python
+obj = cirq.read_json(json_text=string)
+```
+
+## Mechanism
+
+When writing JSON, Cirq checks if the given object has a `_json_dict_` method.
+If it does, the object is replaced by the output of that method.
+Otherwise there are a series of several hardcoded cases for complex numbers, numpy arrays, sympy expressions, and a few others.
```

Comment:
nit: I think it should be "Otherwise, there are..." with a comma.

Reply:
Done.